Conjur/CONTRIBUTING.md

```diff
@@ -118,7 +151,233 @@ important to follow when developing.
+- Fixed issue with dependencies of the project
+
+# Irrelevant to end-user
+- Added more tests around API connectivity ([#123](https://github.com/org/repo/issues/123))
+
+# Bad issue links
+- Added FIPS compatibility to the agent (#123)
+- Added FIPS compatibility to the agent (org/repo#123)
+- Added FIPS compatibility to the agent ([#123](org/repo#123))
+- Added FIPS compatibility to the agent ([org/repo#123](#123))
+...
+```
+
+**Good entries**:
+```markdown
+...
+- Added FIPS compatibility to the agent ([org/repo#123](https://github.com/org/repo/issues/123))
+- Removed `-n` flag from the accepted CLI flags ([org/repo#234](https://github.com/org/repo/issues/234))
+- Improved performance of main API endpoints with change to
+  a new router `foo` package dependency
+- Fixed CVE-2020-12345 by upgrading `testing` dependency to v3 ([org/repo#345](https://github.com/org/repo/issues/345))
+- Fixed overrides handling of `Client` account param [org/repo#456](https://github.com/org/repo/issues/456)
+...
+```
+
+## Tagging
+
+All projects cyclically get a point where they are ready to package a group of changes as a
+check-pointed release which in most cases is started with creating versions through `git tag`.
+Git tags create special branches that allow for a reproducible and easy-to-understand
+delivery of code that are generally 1:1 mapped with significant feature changes.
+
+In most cases for our projects we use [annotated git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_annotated_tags)
+with the name matching the `v<semver>` tri-dotted pattern (e.g. `v1.2.3`). By doing this,
+we make finding relevant code and automating releases easy and simple. Where possible, we
+also use GPG signing of those tags though this process has not been yet fully designed.
+
+To create an annotated tag you have to:
+- **Check out the default branch (in most cases this is `master`) and/or the commit that you
+  want to tag**. In almost all cases, tags should be created on the default branch for
+  automation purposes as a lot of our deliverables get pushed only when the default branch
+  has a specific tag. This checkout step is somewhat optional as `git tag` can also take
+  in a commit to tag but this procedure is less error-prone vs direct tagging.
+
+  Example of checking out the current upstream main branch history
+  ```sh-session
+  $ git fetch
+  $ git checkout master
+  $ git reset --hard origin/master
+  ```
+- **Verify that the git history looks as expected**. Tools like `git log`, `tig`, `gitk`, and
+  others can help you verify that the history is in the expected state.
+- **Create the annotated tag**. To create the tag, use the `git tag -a` command. The main tag
+  **must** be in `v<semver>` tri-dotted format:
+  ```sh-session
+  $ # If you don't have a GPG key set up
+  $ git tag -a "v1.2.3"
+
+  $ # If you do have a GPG key set up
+  $ git tag -s "v1.2.3"
+
+  $ # Make sure to add a commit header and description when the editor window opens
```

Comment:
I always make my message simply `v1.2.3`, and leave the additional description to the github release - wdyt about giving that general instruction here? if you think more detail is required, is there an example header / description that you'd add?